presentation/src/main/java/org/sagebionetworks/research/presentation/speech/TextToSpeechService.kt

```diff
@@ -0,0 +1,437 @@
+    var speechGracePeriod: Long = 5
+    // The public getter for state should have type live data so client's don't modify it.
+    val state: LiveData<TextToSpeechState>
+        get() = _state
+    // The private state value is mutable so the service can update it's state as needed
+    private val _state = MutableLiveData<TextToSpeechState>()
+    private val bindCount = MutableLiveData<Int>()
+    private val cleanupMediator = MediatorLiveData<Pair<Int, TextToSpeechState>>()
+    private val cleanupObserver: Observer<Pair<Int, TextToSpeechState>>
+
+    init {
+        bindCount.postValue(0)
+        _state.postValue(TextToSpeechState(IDLE, null))
+        // We observe the state so that if every client unbinds and the service stops speaking, the service
+        // stops itself.
+        cleanupMediator.postValue(Pair(0, TextToSpeechState(IDLE, null)))
+        cleanupMediator.addSource(bindCount) { count ->
+            // these checks must be null safe to prevent the service from crashing when
+            cleanupMediator.postValue(
+                    Pair(count ?: 0,
+                            cleanupMediator.value?.second ?: TextToSpeechState(IDLE, null)))
+        }
+        cleanupMediator.addSource(_state) { s ->
+            cleanupMediator.postValue(Pair(cleanupMediator.value?.first ?: 0,
+                    s ?: TextToSpeechState(IDLE, null)))
+        }
+        cleanupObserver = Observer { pair ->
+            if (pair != null && pair.first == 0 && pair.second.speakingState != SPEAKING
+                    && pair.second.speakingState != QUEUED) {
+                stopSelf()
+            }
+        }
+
+        cleanupMediator.observeForever(cleanupObserver)
+    }
+
+    override fun onCreate() {
+        super.onCreate()
+        AndroidInjection.inject(this)
+    }
+
+    override fun onBind(intent: Intent?): IBinder {
+        bindCount.postValue((bindCount.value ?: 0) + 1)
+        if (LOGGER.isDebugEnabled) {
+            LOGGER.debug("onBind() called bindCount = $bindCount")
+        }
+
+        return serviceBinder
+    }
+
+    override fun onUnbind(intent: Intent?): Boolean {
+        bindCount.postValue(bindCount.value!! - 1)
+        if (LOGGER.isDebugEnabled) {
+            LOGGER.debug("onUnbind() called bindCount = $bindCount")
+        }
+
+        return super.onUnbind(intent)
+    }
+
+    override fun onDestroy() {
+        textToSpeech?.stop()
+        textToSpeech?.shutdown()
+        textToSpeech = null
+        cleanupMediator.removeObserver(cleanupObserver)
+        clear()
+        super.onDestroy()
+    }
+
+    override fun onInit(status: Int) {
+        if (status == TextToSpeech.SUCCESS) {
+            val languageAvailable = textToSpeech!!.isLanguageAvailable(Locale.getDefault())
+            // >= 0 means LANG_AVAILABLE, LANG_COUNTRY_AVAILABLE, or LANG_COUNTRY_VAR_AVAILABLE
+            if (languageAvailable >= 0) {
+                textToSpeech!!.language = Locale.getDefault()
```

Comment:
At this point you should probably do a let call to avoid the "!!"...

textToSpeech?.let {
   // here you reference textToSpeech as "it" and it won't be optional
}